pkg/jx/cmd/install.go

```diff
@@ -665,251 +693,209 @@ func (options *InstallOptions) Run() error {
-			options.enableOpenShiftRegistryPermissions(ns, helmConfig, dockerRegistry)
-		}
+	if err != nil {
+		return errors.Wrap(err, "failed to install/upgrade the jenkins-x platform chart")
 	}
 
-	if initOpts.Flags.TillerNamespace != "" {
-		if helmConfig.Jenkins.Servers.Global.EnvVars == nil {
-			helmConfig.Jenkins.Servers.Global.EnvVars = map[string]string{}
-		}
-		helmConfig.Jenkins.Servers.Global.EnvVars["TILLER_NAMESPACE"] = initOpts.Flags.TillerNamespace
-		os.Setenv("TILLER_NAMESPACE", initOpts.Flags.TillerNamespace)
+	err = options.waitForInstallToBeReady(namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to wait for jenkinx-x chart installation to be ready")
 	}
-	isProw := false
-	if options.Flags.GitOpsMode {
-		isProw = options.Flags.Prow
-	} else {
-		options.SetDevNamespace(ns)
-		isProw, err = options.isProw()
-		if err != nil {
-			return errors.Wrapf(err, "cannot work out if this is a prow based install in namespace %s", options.currentNamespace)
-		}
+	log.Infof("Jenkins X deployments ready in namespace %s\n", namespace)
+	return nil
+}
+
+func (options *InstallOptions) installPlatformGitOpsMode(gitOpsEnvDir string, gitOpsDir string, configStore configio.ConfigStore,
+	chartRepository string, chartName string, namespace string, version string, valuesFiles []string, secretsFiles []string) error {
+	options.CreateEnvOptions.NoDevNamespaceInit = true
+	deps := []*helm.Dependency{
+		{
+			Name:       JenkinsXPlatformChartName,
+			Version:    version,
+			Repository: DEFAULT_CHARTMUSEUM_URL,
+		},
+	}
+	requirements := &helm.Requirements{
+		Dependencies: deps,
 	}
 
-	if isProw {
-		enableJenkins := false
-		helmConfig.Jenkins.Enabled = &enableJenkins
+	chartFile := filepath.Join(gitOpsEnvDir, helm.ChartFileName)
+	requirementsFile := filepath.Join(gitOpsEnvDir, helm.RequirementsFileName)
+	secretsFile := filepath.Join(gitOpsEnvDir, helm.SecretsFileName)
+	valuesFile := filepath.Join(gitOpsEnvDir, helm.ValuesFileName)
+	err := helm.SaveRequirementsFile(requirementsFile, requirements)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to save GitOps helm requirements file %s", requirementsFile)
 	}
 
-	// lets add any GitHub Enterprise servers
-	gitAuthCfg, err := options.CreateGitAuthConfigService()
+	err = configStore.Write(chartFile, []byte(GitOpsChartYAML))
 	if err != nil {
-		return errors.Wrap(err, "failed to create the git auth config service")
+		return errors.Wrapf(err, "Failed to save file %s", chartFile)
 	}
-	err = options.addGitServersToJenkinsConfig(helmConfig, gitAuthCfg)
+
+	err = helm.CombineValueFilesToFile(secretsFile, secretsFiles, JenkinsXPlatformChartName, nil)
 	if err != nil {
-		return errors.Wrap(err, "failed to add the Git servers to Jenkins config")
+		return errors.Wrapf(err, "Failed to generate %s by combining helm Secret YAML files %s", secretsFile, strings.Join(secretsFiles, ", "))
```

Comment:
nit: I think errors in go are supposed to be lowercase (I don't really mind anyway)